packages/logging/src/analytics-helpers.ts

```diff
@@ -378,3 +378,34 @@ export class ThrottledAnalytics implements MongoshAnalytics {
     });
   }
 }
+
+type SampledAnalyticsOptions = {
+  target?: MongoshAnalytics;
+  sampling: () => boolean;
+};
+
+export class SampledAnalytics implements MongoshAnalytics {
+  private isEnabled: boolean;
+  private target: MongoshAnalytics;
+
+  constructor(configuration: SampledAnalyticsOptions) {
+    this.isEnabled = configuration.sampling();
+    this.target = configuration.target || new NoopAnalytics();
+  }
+
+  get enabled(): boolean {
+    return this.isEnabled;
+  }
+
+  identify(message: AnalyticsIdentifyMessage): void {
+    this.isEnabled && this.target.identify(message);
+  }
+
+  track(message: AnalyticsTrackMessage): void {
+    this.isEnabled && this.target.track(message);
+  }
+
+  flush(callback: (err?: Error | undefined) => void): void {
```

Comment:
@addaleax in case the telemetry is disabled, what should we do? ToggleableAnalytics it still runs the flush and the callback even if it's toggled off. I suppose it should be fine to do the same here, as the event queue should be empty.

Reply:
Yes, I think that’s fine – shouldn’t be doing anything in this case :)